src/neighborhoods/ellipsoid.jl

```diff
@@ -0,0 +1,45 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    EllipsoidNeighborhood(semiaxes, angles; convention=:TaitBryanExtr)
+
+An ellipsoid neighborhood with `semiaxes` and `angles`. For 2D ellipses,
+there are two semiaxes and one rotation angle. For 3D ellipsoids, there are
+three semiaxes and three rotation angles. Different rotation conventions can be
+passed via the `convention` keyword argument. The list of conventions is
+available in the [aniso2distance](@ref) documentation.
+"""
+struct EllipsoidNeighborhood <: AbstractBallNeighborhood
+  semiaxes::AbstractVector
+  angles::AbstractVector
+  convention::Symbol
+  metric::Mahalanobis
```

Comment:
I think we need to add type parameters here to avoid abstract types. For example:

```julia
struct EllipsoidNeighborhood{S,A,C,M} <: AbstractBallNeighborhood
  semiaxes::S
  angles::A
  convention::C
  metric::M
end
```